app/src/main/java/io/appium/uiautomator2/handler/GetElementAttribute.java

```diff
@@ -6,33 +6,44 @@
 import android.support.test.uiautomator.UiObject;
 import android.support.test.uiautomator.UiObject2;
 import android.support.test.uiautomator.UiObjectNotFoundException;
-import android.support.test.uiautomator.UiSelector;
 import android.view.accessibility.AccessibilityNodeInfo;
 
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import java.io.InvalidClassException;
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import java.text.MessageFormat;
 
-import io.appium.uiautomator2.common.exceptions.InvalidSelectorException;
 import io.appium.uiautomator2.common.exceptions.NoAttributeFoundException;
 import io.appium.uiautomator2.common.exceptions.UiAutomator2Exception;
 import io.appium.uiautomator2.core.AccessibilityNodeInfoGetter;
+import io.appium.uiautomator2.core.EventRegister;
+import io.appium.uiautomator2.core.ReturningRunnable;
+import io.appium.uiautomator2.core.UiObjectChildGenerator;
 import io.appium.uiautomator2.handler.request.SafeRequestHandler;
 import io.appium.uiautomator2.http.AppiumResponse;
 import io.appium.uiautomator2.http.IHttpRequest;
+import io.appium.uiautomator2.model.AccessibilityScrollData;
 import io.appium.uiautomator2.model.AndroidElement;
+import io.appium.uiautomator2.model.AppiumUiAutomatorDriver;
 import io.appium.uiautomator2.model.KnownElements;
 import io.appium.uiautomator2.model.UiObject2Element;
 import io.appium.uiautomator2.server.WDStatus;
 import io.appium.uiautomator2.utils.Device;
 import io.appium.uiautomator2.utils.Logger;
-import io.appium.uiautomator2.utils.ReflectionUtils;
+
+import static io.appium.uiautomator2.utils.Device.getUiDevice;
 
 public class GetElementAttribute extends SafeRequestHandler {
 
+    // these constants are magic numbers experimentally determined to minimize flakiness in generating
+    // last scroll data used in getting the 'contentSize' attribute.
+    // TODO see whether anchoring these to time and screen size is more reliable across devices
+    private static int MINI_SWIPE_STEPS = 10;
+    private static int MINI_SWIPE_PIXELS = 600;
```

Comment:
Why you name this "MINI"? It's a half of 1280x720 screen.

Reply:
you're right. it was supposed to be MINI but i found with a small swipe the accessibility event was not generated. i think listview elements actually need to cross indexes in order to generate the event. but this could also be related to notification listener flakiness, so will experiment again with that turned off.